Lecture 7 HW

Bepely opened the pull request:
• Added react-redux and RTK libraries
• Mock data Replaced with Normalized mock data
• Created a Store and slices for normalized data
• Replaced data handling from regular data to RTK

__

С трудом вник в механизм переноса приложения на RTK с обычного объекта с данными. 
Есть ощущение, что я наворотил чего-то очень неправильного

Bepely committed:
Lecture 7 HW

• Added react-redux and RTK libraries
• Mock data Replaced with Normalized mock data
• Created a Store and slices for normalized data
• Replaced data handling from regular data to RTK

zubkov7 commented on src/restarauntWrapper/component.jsx:
```diff
@@ -0,0 +1,31 @@
+    return (
+        <>
+            <RestaurantTabsContainer currentRestaurant={currentRestaurant} setCurrentRestaurantId={setCurrentRestaurantId} />
+            <Restaurant currentRestaurant={currentRestaurant} />
```
в подходе с редаксом обычно передают id, а сам компонент уже достает из редакса полную сущность

zubkov7 commented on src/restarauntWrapper/component.jsx:
```diff
@@ -0,0 +1,31 @@
+
+    return (
+        <>
+            <RestaurantTabsContainer currentRestaurant={currentRestaurant} setCurrentRestaurantId={setCurrentRestaurantId} />
```
в случае с табами мы тоже можем пробросить список айдишников и айди активного ресторана, таб ресторана достанет по айди имя ресторана

zubkov7 commented:
в целом подход такой: прокидываем список айди, а сами компоненты достают сущность необходимую

Bepely committed:
lecture-7-hw-2

• Reorganized file structure to have a components folder
• Improved RTK data usage. (from props drilling to selectors)

zubkov7 commented on src/components/menu/component.jsx:
```diff
@@ -2,15 +2,16 @@ import MenuItem from "../menuItem/component";
-            {menu.map((element) => {
-                return <MenuItem menuItem={element} />
+            {menuDishesIds.map((dishId) => {
+                return <MenuItem dishId={dishId} />
```
необходимо добавить проп key

zubkov7 commented on src/components/menuItem/component.jsx:
```diff
@@ -0,0 +1,21 @@
+            // Fetch the dishes based on the menu array
+            const dish = useSelector((state) => selectDishById(state, dishId));
+
+    return <li className={styles.menuItem} key={dishId}>{dish.name} 
```
теги li лучше использовать в месте рендера массива, так как сам компонент может существовать отдельно

zubkov7 commented on src/components/restarauntTabsContainer/component.jsx:
```diff
@@ -0,0 +1,30 @@
+    const restaurantIds = useSelector(selectRestarantsIds);
+
+    // Map over the restaurant IDs to get the actual restaurant objects
+    const restaurants = useSelector((state) =>
```
в подходе с редаксом в компоненты передается айдишник, а он сам уже использует селектор, чтобы достать данные сущности

zubkov7 commented on src/components/review/component.jsx:
```diff
@@ -1,8 +1,14 @@
-function Review({reviews}) {
+function Review({reviewIds}) {
+
+    const reviews = useSelector((state) => 
```
тут аналогично

Bepely committed:
lecture-7-hw-3

• Added key props
• Reworked<li> element usage
• Reworked redux selectors usage

zubkov7 commented on src/components/menu/component.jsx:
```diff
@@ -0,0 +1,25 @@
+        {menuDishesIds?.length && (
+            <ul className={styles.listWrapper}>
+            {menuDishesIds.map((dishId) => {
+                return <li key={`menu-item-li-${dishId}`} className={styles.menuItem}>
```
для key необязательно добавлять префикс. Главное, чтобы он был уникален среди соседей при рендере списка

key нужно добавлять только для элемента, который оборачивает, то есть для MenuItem уже не нужно

Bepely committed:
lecture-7-hw-4

• Updated key prop usage
